### PR TITLE
Marco/feature/buttons

### PR DESCRIPTION
**Pull request by MarcoFrancisco13**
- [x] Long and Short Button classes
- [x] Select Buttons for Date, Semester, and Quarter
- [ ] Fixing images (using Image.asset) 
- [ ] Dropdown Select for Grades  

image

### CODE REVIEW

File: lib/widgets/qpi/buttons.dart [hunk 1]
```diff
@@ -0,0 +1,56 @@
+import 'package:flutter/material.dart';
```
Comment (devnamme) [inline lib/widgets/qpi/buttons.dart]: move file to `lib/widgets/` for clarity; these buttons are used in almost all pages and not just by the qpi

File: lib/widgets/qpi/buttons.dart [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+class LongButton extends StatefulWidget {
+  final String text;
+  final Color buttonColor;
+  final Color textColor;
+LongButton(this.text, this.buttonColor, this.textColor);
```
Comment (devnamme) [inline lib/widgets/qpi/buttons.dart]: for the `onPressed`, you can add another variable here

`final VoidCallback onPressed` and add it in the parameter list
source [here](https://www.digitalocean.com/community/tutorials/flutter-widget-communication)

to add it in the page itself, you can do:
```dart
// ...
children: [
  LongButton(text: 'hello', buttonColor: Colors.white, textColor: Colors.black,
    onPressed: () {
      print('hello world');
    },
  ),
// ...
```
i haven't tested it yet but it *should technically* work - let me know if something's wrong

File: lib/widgets/qpi/buttons.dart [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+  @override
+  Widget build(BuildContext context) {
+    return Container(
+      width: 219,
+      height: 56,
+      decoration: BoxDecoration(
+          color: widget.buttonColor,
+          borderRadius: BorderRadius.all(Radius.circular(5))),
+      child: TextButton(
+        onPressed: () {},
```
Comment (devnamme) [inline lib/widgets/qpi/buttons.dart]: call `onPressed()` here or whatever you named the variable in the widget

File: lib/widgets/qpi/buttons.dart [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+class _LongButtonState extends State<LongButton> {
+  @override
+  Widget build(BuildContext context) {
+    return Container(
+      width: 219,
+      height: 56,
+      decoration: BoxDecoration(
+          color: widget.buttonColor,
+          borderRadius: BorderRadius.all(Radius.circular(5))),
```
Comment (devnamme) [inline lib/widgets/qpi/buttons.dart]: i'm not sure if `5` is the border (and with the width of `219` rin), most should be a factor of 8 already but clarify na lang with designers

File: lib/widgets/qpi/selectbutton.dart [hunk 5]
```diff
@@ -0,0 +1,271 @@
...
+                style: TextStyle(color: widget.labelColor),
+              )
+            ],
+          ),
+          Row(
+            mainAxisAlignment: MainAxisAlignment.spaceBetween,
+            children: [
+              // I think a listview builder might be a better way to create this but can't seem to make it work yet
+
+              Container(
```
Comment (devnamme) [inline lib/widgets/qpi/selectbutton.dart]: you can convert this into a row of `SquareButton`s (new class)

**Comment by devnamme** [comment]
ideally there should only be four classes:
ShrinkingButton (or smth else, for the quarter)
SquareButton
ShortButton
LongButton

and then the rest (entire quarter select, semester select, etc.) are in the pages

`Row of SquareButtons with spaceBetween` for the day select and sem select
`Row of ShrinkingButtons with spaceBetween` for the quarter select

but ofc, it's up to you if you want to make the entire thing a single widget. probably edit this in the next sprint itself?

**Comment by MarcoFrancisco13** [comment]
noted! I'll try to make edits on this based on your comments as soon as I can. Thanks for the review!

**Comment by devnamme** [comment]
oops i just noticed that the heights of the input boxes and the selector buttons are different (54 and 56) while the square button components use 56. looks like those are different - i'll let the designers know

**Comment by MarcoFrancisco13** [comment]
- [x] Dropdown Button
- [x] Using 4 primary classes for buttons.dart (Long, Short, Square, Shrinking)
- [x] Moved buttons.dart to lib/widgets 

image
image

Clearing this backlog so I can begin the next sprint, sorry for the delay!

**Review by devnamme — APPROVED** [review]
hello! are you going to edit the image assets here in this branch or in a separate branch?

File: lib/widgets/buttons.dart [hunk 6]
```diff
@@ -0,0 +1,453 @@
...
+  final Color buttonColor;
+  final Color textColor;
+  final VoidCallback onPressed;
+
+  SquareButton(this.text, this.buttonColor, this.textColor, this.onPressed);
+  @override
+  Widget build(BuildContext context) {
+    return Container(
+      width: 54,
+      height: 54,
```
Comment (devnamme) [inline lib/widgets/buttons.dart]: (new) from design meeting: switch to 56x56

File: lib/widgets/buttons.dart [hunk 7]
```diff
@@ -0,0 +1,453 @@
...
+  final Color buttonColor;
+  final Color textColor;
+  final VoidCallback onPressed;
+
+  ShrinkingButton(this.text, this.buttonColor, this.textColor, this.onPressed);
+  @override
+  Widget build(BuildContext context) {
+    return Container(
+      width: 41,
+      height: 54,
```
Comment (devnamme) [inline lib/widgets/buttons.dart]: switch to height 56

File: lib/widgets/buttons.dart [hunk 8]
```diff
@@ -0,0 +1,453 @@
...
+}
+
+class _DropDownState extends State<DropDown> {
+  int _value = 1;
+
+  @override
+  Widget build(BuildContext context) {
+    return Container(
+      width: 185,
+      height: 54,
```
Comment (devnamme) [inline lib/widgets/buttons.dart]: height 56

for the width, im not sure if it should really be 185 (or it expands) but that can still work

**Comment by MarcoFrancisco13** [comment]
> 
> 
> hello! are you going to edit the image assets here in this branch or in a separate branch?

I'll be doing it along with the branch for the next sprint! I've also fixed the heights now so if it's all ready I'll fix the conflicting files to merge, thanks!

**Review by devnamme — APPROVED** [review]
alright, good for merging! i'll let you fix the conflicts, sorry for the late review

good job btw!

**Pull request merged**